scipy/stats/_continuous_distns.py

```diff
@@ -2306,6 +2306,53 @@ def _entropy(self, c):
 gompertz = gompertz_gen(a=0.0, name='gompertz')
 
 
+class gamma_gompertz_gen(rv_continuous):
+    """ A Gamma/Gompertz continuous random variable
+    
+    %(before_notes)s
+    
+    Notes
+    -----
+    The probability density function for `gamma_gompertz` is::
+    
+        gamma_gompertz.pdf(x, c, beta) = c * exp(x) * (beta**c) / ((beta - 1 + exp(x))**(c + 1))
+    
+    for ``x >= 0``, ``c > 0``, ``beta > 0``.
+    
+    `gamma_gompertz` takes ``c`` and ``beta`` as shape parameters
+    
+    %(after_notes)s
+    
+    References
+    ----------
+    [1] https://en.wikipedia.org/wiki/Gamma/Gompertz_distribution    
+    
+    %(example)s
+    """
+        
+    def _logsf(self, x, c, beta):
+        # np.where is used for dealing with the case that x is too large
+        # so that sc.expm1(x) becomes infinity.
+        # Since x is very large in that case, log(beta + expm1(x))
+        # approximately equals to x
+        return c * (np.log(beta) - np.where(np.isfinite(sc.expm1(x)),
+                                            np.log(beta + sc.expm1(x)), x)) 
```

Comment:
This computes `sc.expm1(x)` twice. Suggest computing it once, storing in a local variable and using the local variable twice.
The expression `-c*np.log1p(np.expm1(x)/beta)` may be equivalent and only involve two transcendental function evaluations for the common case. (Again storing `np.expm1(x)` in a local variable, and using `-c*(x-np.log(beta))` in the uncommon case.)

Reply:
Thanks for your suggestion. Accordingly,
```
        # np.where is used for dealing with the case that x is too large
        # so that sc.expm1(x) becomes infinity.
        # Since x is very large in that case, log(beta + expm1(x))
        # approximately equals to x
        return c * (np.log(beta) - np.where(np.isfinite(sc.expm1(x)),
                                            np.log(beta + sc.expm1(x)), x))
```
to
```
        em1 = sc.expm1(x)
        # np.where is used for dealing with the case that x is too large
        # so that sc.expm1(x) becomes infinity.
        # Since x is very large in that case, -sc.log1p(em1 / beta)
        # approximately equals to -sc.log(np.exp(x) / beta) = np.log(beta) - x
        return c * (np.where(np.isfinite(em1), -sc.log1p(em1 / beta),
                             np.log(beta) - x))
```